app/knights_battle/battle.py

```diff
@@ -0,0 +1,26 @@
+from app.knights_battle.knight import Knight
+
+
+def battle(knights_config: dict) -> dict:
+    result = {}
+
+    for element in knights_config:
+        knight = Knight(name="", power=0, hp=0, protection=0)

```

Comment:
Pass base parameters each of knights instead of providing some default values to omit `adding_value_to_instance` method